CMakeLists.txt

```diff
@@ -308,6 +308,14 @@ option(USE_DISTRIBUTED "Use distributed" ON)
 cmake_dependent_option(
     USE_MPI "Use MPI for Caffe2. Only available if USE_DISTRIBUTED is on." ON
     "USE_DISTRIBUTED" OFF)
+cmake_dependent_option(
+    USE_UCC "Use UCC. Only available if USE_DISTRIBUTED is on." OFF
+    "USE_DISTRIBUTED" OFF)
+cmake_dependent_option(
+    USE_SYSTEM_UCC "Use system-wide UCC" OFF
+    "USE_UCC" OFF)
```

Comment:
Yep, the plan is to enable UCC gradually. As of this PR, everything related to UCC is default to OFF, so that it won't break any existing things. Once we port over the tests and harden them, we can start to gradually switch the UCC knobs to default ON.